protocol/contracts/nexus/Shard.sol

```diff
@@ -0,0 +1,489 @@
+        }
+        _mintBatch(to, shardIds, amounts, "");
+    }
+
+    /*
+     * @notice Burn batch shards. Overriden from base contract. 
+     * Modified to allow Relic contract to burn blacklisted account shards.
+     * @param account The account owning the shard
+     * @param ids The shard IDs
+     * @param values The amounts of each shard to burn
+     */
+    function burnBatch(address account, uint256[] memory ids, uint256[] memory values) public override {
+        // allow relic to burn blacklisted relic shards
+        if (_msgSender() == address(relic)) {
+            _burnBatch(account, ids, values);
+            return;
+        } else if (account != _msgSender() && !isApprovedForAll(account, _msgSender())) {
+            revert ERC1155MissingApprovalForAll(_msgSender(), account);
+        }
+        _burnBatch(account, ids, values);
+    }
+
+    /*
+     * @notice Get shard IDs a minter is allowed to mint
+     * @param minter The minter with perms
+     * @return Shard IDs array
+     */
+    function getMinterAllowedShardIds(address minter) external view returns (uint256[] memory) {
+        return allowedShardIds[minter].values();
+    }
+
+    function nextTokenId() external view returns (uint256) {
+        return _nextTokenId();
+    }
+
+    /*
+     * @notice Check if given ID is a valid shard ID
+     * @param id The ID to check
+     * @return If ID is a valid shard
+     */
+    function isShardId(uint256 id) external view returns (bool) {
+        return _startTokenId() <= id && id < _nextTokenId();
+    }
+
+    /*
+     * @notice Get information about a recipe
+     * @param recipeId The ID of the recipe
+     * @return Recipe information struct. see above
+     */
+    function getRecipeInfo(uint256 recipeId) external view returns (Recipe memory info) {
+        info = recipes[recipeId];
+    }
+
+    function nextRecipeId() external view returns (uint256) {
+        return _currentRecipeIndex;
+    }
+
+    /*
+     * @notice Get shard IDs of an enclave
+     * @param enclave The enclave
+     * @return Shard IDs of enclave
+     */
+    function getEnclaveShards(Enclave enclave) external view returns (uint256[] memory) {
+        return enclaveShards[uint8(enclave)].values();
+    }
+
+    /*
+     * @notice Determines if a shard ID belongs to an enclave
+     * @param enclave The enclave
+     * @param shardId The shard ID
+     * @return True if shard ID belongs to enclave, else false.
+     */
+    function isEnclaveShard(Enclave enclave, uint256 shardId) external view returns (bool) {
+        return enclaveShards[uint8(enclave)].contains(shardId);
+    }
+
+    /*
+     * @notice Get the information of a minter. 
+     * Fucntion is not validating minter and reverting. Caller should check and handle zero values
+     * @param minter The minter
+     * @return MintInfo struct information of minter
+     */
+    function getMintInfo(address minter) external view returns (MintInfo memory info) {
```

Comment:
An alternative struct which would be easier for the client to digest would be:
```
struct MintInfo {
    uint256 shardId;
    uint256 balance;
    uint256 cap;
}

function getMintInfo(address minter) external view returns (MintInfo[] memory info) {
    EnumerableSet.UintSet storage minterShards = allowedShardIds[minter];
    uint256 _length = minterShards.length();
    info = new MintInfo[](_length);
    for (uint256 i; i < _length;) {
        shardId = minterShards.at(i);
        info[i] = MintInfo({
            shardId: shardId,
            balance: mintBalances[minter][shardId],
            cap: allowedShardCaps[minter][shardId]
        });

        unchecked {
            ++i;
        }
    }
}
```

Reply:
good one